src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/DockerProcessor.kt

```diff
@@ -32,12 +31,13 @@ class DockerProcessor {
                 .build()
 
         val creation = docker.createContainer(containerConfig)
-        containerId = creation.id()
+        val containerId = creation.id()
         docker.startContainer(containerId)
 
         val hash = docker.logs(containerId, DockerClient.LogsParam.stdout()).readFully()
-        val spaceIndex = hash.indexOfFirst { it.isWhitespace() }
+        docker.stopContainer(containerId, 0)
+        docker.removeContainer(containerId)
```

Comment:
Please add try-catch-finally blocks and put removeContainer into finally. And print exception stack traces.

Reply:
What value does variable containerId should be initialized?
There must be a value, but if I put null, in that case, removing null container isn't great.

Also what I need to return after try-catch block?

Reply:
Well, you can check if `containerId` is null or not, that's not a big issue.

Return what you do return now in case when exception is not thrown, return null or empty string or throw your own exception or replace String value with some sort of Status with error code. There are many options, actually, and at this point it doesn't matter which one do you choose. What matters is that you know that something may fail.

Reply:
Ok, done